packages/ember-glimmer/tests/utils/shared-conditional-tests.js

```diff
@@ -374,11 +384,56 @@ export class SharedConditionalsTest extends AbstractConditionalsTest {
     this.assertText('T1F2');
   }
 
+  ['@test it maintains DOM stability when condition changes from a truthy to a different truthy value']() {
+    this.renderValues(true);
+
+    this.assertText('T1');
+
+    this.takeSnapshot();
+
+    this.runTask(() => set(this.context, 'cond1', 'hello'));
+
+    this.assertText('T1');
+
+    this.assertInvariants();
+  }
+
+  ['@test it maintains DOM stability when condition changes from a falsy to a different falsy value']() {
+    this.renderValues(false);
+
+    this.assertText('F1');
+
+    this.takeSnapshot();
+
+    this.runTask(() => set(this.context, 'cond1', ''));
+
+    this.assertText('F1');
+
+    this.assertInvariants();
+  }
+}
+
+export class SharedHelperConditionalsTest extends SharedConditionalsTest {
+  renderValues(...values) {
+    let templates = [];
+    let context = {};
+
+    for (let i = 1; i <= values.length; i++) {
+      templates.push(this.templateFor({ cond: `cond${i}`, truthy: `t${i}`, falsy: `f${i}` }));
```

Comment:
we can't use `renderValues` from `SharedConditionalsTest` as it uses 

```
{ cond: `cond${i}`, truthy: `{{t}}${i}`, falsy: `{{f}}${i}` }
```

and that would produce a template such as:

```
<div data-foo="{{if cond1 {{t}}1 {{f}}1}}" />
```